reconcile/utils/saasherder.py

```diff
@@ -344,6 +364,62 @@ def _validate_image_tag_not_equals_ref(
             )
             self.valid = False
 
+    @staticmethod
+    def _get_upstream_jobs(
+            jjb: JJB,
+            all_jobs: Mapping[str, dict],
+            url: str,
+            ref: str,
+    ) -> Iterable[UpstreamJob]:
+        results = []
+        for instance, jobs in all_jobs.items():
+            for job in jobs:
+                job_repo_url = jjb.get_repo_url(job)
+                if url != job_repo_url:
+                    continue
+                job_ref = jjb.get_ref(job)
+                if ref != job_ref:
+                    continue
+                results.append(UpstreamJob(instance, job['name']))
+        return results
+
+    def validate_upstream_jobs(
+            self,
+            jjb: JJB,
+    ):
+        all_jobs = jjb.get_all_jobs(job_types=["build"])
+        pattern = r'^[0-9a-f]{40}$'
+        for sf, rt, t in self:
+            sf_name = sf['name']
+            rt_name = rt['name']
+            url = rt['url']
+            ref = t['ref']
+            if re.search(pattern, ref):
```

Comment:
is the purpose of this to exclude build jobs with repo refs that are SHAs? that implies we have such configs. what is the story about this?

Reply:
we don't have such configs (#2136), and this is why we "filter" these out.

Reply:
oh sry i read that wrong. ref is from the target and not from the job. that makes sense in regards to #2136